### ISSUE TITLE
Simplify conditonal build steps

### ISSUE THREAD

**Issue by epage** [crate-ci/meta#4]
Examples
- Build step on only on a given channel
- Build step if a env variable is set to X

**Comment by epage** [comment]
`cargo-when` is a good candidate for this
```bash
cargo when --channel stable,beta build
```

Unfortunately, it doesn't seem to be getting any love.  Desired work
- pre-built binaries
- environment support

I reached out to the author for joining effort (https://github.com/starkat99/cargo-when/issues/2) but haven't heard back.

The repo hasn't been touched in 2 years but the author is still active on github.

**Comment by epage** [comment]
CAD97 pointed this out to me, seems interesting: http://cs.furman.edu/~cdurham/experiments/Rust/ci_detective/

@CAD97 should some of the conditional steps you've adding to `ghp-upload` be integrated into this tool?

**Comment by CAD97** [comment]
Yes, that's why I'm writing it :) (that's my school webserver, I also could have put it on cad97.com but didn't feel like it)

It's still very much in experimentation phase. But it should allow further CI tools to build on top of it without having to do the doc-crawling of various CI environment setups that I'm doing right now.

**Comment by epage** [comment]
Just to be clear, the idea here is make a command line tool (one exists; trying to reuse it)   So instead of baking the CI policy into `ghp-upload` for when to run, you pass conditions to this wrapper.  it then decides when to invoke your tool.
